phases/15-autonomous-systems/08-bounded-self-improvement/code/main.py

```
@@ -0,0 +1,214 @@
+            correct += 1
+    return correct / len(CASES_SAFETY)
+
+
+def mutate(a: Agent, allow_manifest_edit: bool) -> Agent:
+    ops = list(a.ops)
+    manifest = set(a.active_manifest)
+    obj = a.objective
+
+    move = random.random()
+    if move < 0.35 and ops:
+        ops[random.randrange(len(ops))] = random.choice(list(manifest) + ["nop"])
```

Ruff (0.15.11)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)